.github/workflows/Umbrella_Sandbox_Setup.yml

```diff
@@ -0,0 +1,107 @@
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+###############################################################
+
+name: Umbrella Sandbox Setup
+
+on:
+  #for testing purpouses
+  pull_request:
+    branches:
+      - 'main'
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+  setup-sandbox:
+    runs-on: ubuntu-latest
+    env:
+      NODE_IMAGE: ${{ github.event.inputs.node_image || 'kindest/node:v1.27.3' }}
+      HELM_VERSION: ${{ github.event.inputs.helm_version || 'latest' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Kubernetes KinD Cluster
+        uses: container-tools/kind-action@0ad70e2299366b0e1552c7240f4e4567148f723e
+        with:
+          version: v0.20.0
+          node_image: ${{ env.NODE_IMAGE }}
+
+      - name: Enable Ingress in KinD
+        run: |
+          kubectl apply -f https://kind.sigs.k8s.io/examples/ingress/deploy-ingress-nginx.yaml
+          kubectl wait --namespace ingress-nginx \
+            --for=condition=ready pod \
+            --selector=app.kubernetes.io/component=controller \
+            --timeout=90s
+
+      - name: Verify Cluster
+        run: |
+          kubectl cluster-info
+          kubectl get nodes -o wide
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v4.1.0
+        with:
+          version: ${{ env.HELM_VERSION }}
+
+      - name: Build portal frontend
+        uses: docker/build-push-action@v4.1.0
+        with:
+          context: .
+          file: ./.conf/Dockerfile.full
+          push: true
+          tags: kind-registry:5000/portal-frontend:testing
+
+      - name: Add Tractus-X Helm Repository
+        run: |
+          helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
+          helm repo update
+
+      - name: Install Umbrella Chart Portal Subset
+        run: |
+          helm install \
+            --set portal.enabled=true,centralidp.enabled=true,sharedidp.enabled=true umbrella tractusx-dev/umbrella \
+            --set portal.frontend.image.repository=kind-registry:5000/portal-frontend --set portal.frontend.image.tag=testing \
+            --namespace umbrella --create-namespace
+
+      - name: Verify Deployment
+        run: |
+          kubectl get pods -n umbrella
+          helm ls --namespace umbrella
+
+      - name: Install Node.js for Cypress
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+
+      - name: Install Cypress Dependencies
+        run: |
+          npm install cypress --legacy-peer-deps
```

Comment:
I think this is ready to execute the cypress tests yet, right? could you please have a look at the [cypress docs to gh actions](https://docs.cypress.io/app/continuous-integration/github-actions)

As I mentioned in the last portal sync, I assume that all the addresses in umbrella for portal, centralidp and sharedidp will be needed to be overwritten to the k8s cluster internal services, [example](https://github.com/eclipse-tractusx/tractus-x-umbrella/blob/umbrella-2.8.0/charts/umbrella/values.yaml#L35), because the ingress isn't enabled with the gh actions k8s cluster.